azerothcore/azerothcore-wotlk
# fix(DB): Override Spell Bonus Coefficients (Workaround)

## Changes Proposed:
-  Restore spell_bonus_data prior to https://github.com/azerothcore/azerothcore-wotlk/pull/12562
-  Keeps the code changes so that we are able to continue to work on fix individual spells, but allows them to scale properly until proper fixes are implemented
- Also fixes lower rank spells by applying the correct coefficients extracted from the DBC where possible

## Issues Addressed:

- Does not properly solve any issues, but is workaround for https://github.com/azerothcore/azerothcore-wotlk/issues/13102
- And https://github.com/azerothcore/azerothcore-wotlk/issues/13056

## SOURCE:


## Tests Performed:

- SQL runs without errors
- Not tested in game, but should have same behavior as before


## How to Test the Changes:


1. Run SQL update
2. Test spells from linked issues (like Warlock Immolate) to see if spell power scaling is working

## Known Issues and TODO List:


- We will still want to implement proper workarounds for each spell affected by this and remove them from the DB as we can


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
